subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/java/KotlinDocComment.kt

```diff
@@ -0,0 +1,77 @@
+package org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.java
+
+import org.jetbrains.dokka.analysis.java.*
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
+import org.jetbrains.kotlin.kdoc.psi.impl.KDocTag
+
+internal class KotlinDocComment(
+    val comment: KDocTag,
+    val descriptor: DeclarationDescriptor
```

Comment:
Not sure what exactly you mean. This class has a common interface/abstraction `DocComment`, and `KotlinDocComment` is but an implementation for the descriptor kotlin analysis, so naturally it needs the descriptor to implement the interface methods. 

> Otherwise, they have to be copied into a symbols analysis module.

That's perfectly normal, the symbols analysis module is a whole different implementation of the analysis, I don't think they should share any small pieces of code, especially since we plan to remove the descriptor analysis down the road.

On a side note, I've renamed some classes to include the `Descriptor` prefix so that it doesn't clash with the Symbols implementation